### PR TITLE
infra/port: display driver name

### PR DESCRIPTION
**Pull request by christophefontaine**
Add the driver name to the command "show interface"

grcli show interface name <ifname>
driver:  net_ixgbe

### CODE REVIEW

File: modules/infra/cli/port.c [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 static void port_show(const struct gr_api_client *, const struct gr_iface *iface) {
 	const struct gr_iface_info_port *port = (const struct gr_iface_info_port *)iface->info;
+
+	printf("driver:  %s\n", port->driver_name);
```
Comment (rjarry) [inline modules/infra/cli/port.c:21]: Can you move it just below devargs? It seems more natural.

File: modules/infra/control/port.c [hunk 2]
```diff
@@ -439,13 +439,20 @@ static int iface_port_get_eth_addr(const struct iface *iface, struct rte_ether_a
...
 
 	memccpy(api->devargs, port->devargs, 0, sizeof(api->devargs));
 	memcpy(&api->mac, &port->mac, sizeof(api->mac));
 	api->n_rxq = port->n_rxq;
 	api->n_txq = port->n_txq;
 	api->rxq_size = port->rxq_size;
 	api->txq_size = port->txq_size;
+
+	if (rte_eth_dev_info_get(port->port_id, &dev_info) == 0) {
+		strncpy(api->driver_name, dev_info.driver_name, GR_PORT_DRIVER_NAME_SIZE - 1);
```
Comment (rjarry) [inline modules/infra/control/port.c:452]: Please use `memccpy` instead.

File: modules/infra/api/gr_infra.h [hunk 3]
```diff
@@ -53,6 +53,8 @@ struct gr_iface {
 
 // Info for GR_IFACE_TYPE_PORT interfaces
 struct gr_iface_info_port {
+#define GR_PORT_DRIVER_NAME_SIZE 16
```
Comment (rjarry) [inline modules/infra/api/gr_infra.h:56]: Can you check what is the max size for driver names in DPDK?

**Review by rjarry — CHANGES_REQUESTED** [review]

File: modules/infra/cli/port.c (hunk 1 above)
Comment (christophefontaine) [reply modules/infra/cli/port.c]: done

File: modules/infra/api/gr_infra.h (hunk 3 above)
Comment (christophefontaine) [reply modules/infra/api/gr_infra.h]: done

**Review by rjarry — APPROVED** [review]

**Pull request merged**